### PR TITLE
[PR1] DEV-55 Added Transmit Table

### PR DESCRIPTION
**Pull request by smorsua**
This PR adds the transmit table, which is used to send orders to the vehicle. It currently supports checkbox for the boolean values and input for the numeric values. Dropdowns will be added in the future to support enums. Input verification will be added in the future.

[screen-capture (5).webm](https://user-images.githubusercontent.com/114338434/204144997-cc184305-d199-4358-809d-48102ace9cc6.webm)

### CODE REVIEW

File: src/components/PacketTable/TransmitTable/OrderField.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        <TextInput
+          onChange={(value: string | number) => {
+            onChange(true, value);
+          }}
+          checkInput={() => {
+            return true;
+          }}
+        ></TextInput>
+      );
+    case "boolean":
```
Comment (jmaralo) [inline src/components/PacketTable/TransmitTable/OrderField.tsx:45]: this should be just bool

File: src/components/PacketTable/TransmitTable/OrderField.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    case "uint8":
+    case "uint16":
+    case "uint32":
+    case "uint64":
+    case "int8":
+    case "int16":
+    case "int32":
+    case "int64":
+    case "float32":
+    case "float64":
```
Comment (jmaralo) [inline src/components/PacketTable/TransmitTable/OrderField.tsx:34]: this is hard to follow (but we can't avoid it), what I did in the backend was to have all numeric cases as the default case, but this also works

File: src/components/PacketTable/TransmitTable/OrderFields.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+          return (
+            <OrderField
+              key={field.name}
+              name={field.name}
+              valueType={field.valueType}
+              onChange={(
+                isValid: boolean,
+                newValue: string | number | boolean
+              ) => {
+                //FIXME: que sea mas obvio que la id es el name de la orden o hacerlo de otra manera mas explicita
```
Comment (jmaralo) [inline src/components/PacketTable/TransmitTable/OrderFields.tsx:27]: we should try to avoid other languages, even in code (but this is more a personal preference)

**Review by jmaralo — CHANGES_REQUESTED** [review]
There is just one main thing that I found, the rest are just my opinion

**Review by jmaralo — APPROVED** [review]

File: src/components/PacketTable/TransmitTable/OrderFields.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+          return (
+            <OrderField
+              key={field.name}
+              name={field.name}
+              valueType={field.valueType}
+              onChange={(
+                isValid: boolean,
+                newValue: string | number | boolean
+              ) => {
+                //FIXME: make it more obvious that the id is the name of the order or do it another way
```
Comment (lipezaballa) [inline src/components/PacketTable/TransmitTable/OrderFields.tsx:27]: I would like to add some comments to my code for future improvements, is this the way to do it?

File: src/components/PacketTable/TransmitTable/TransmitTable.tsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+import { RootState } from "store";
+
+export const TransmitTable = () => {
+  const orderService = useContext(OrderServiceContext);
+  const orderDescriptions = useSelector((state: RootState) => {
+    return state.orders;
+  });
+
+  const createSendOrder = (orderDescription: OrderDescription) => {
+    return (fields: FieldState[]) => {
```
Comment (lipezaballa) [inline src/components/PacketTable/TransmitTable/TransmitTable.tsx:19]: Which is the function of the fieldstate?

File: src/styles/globalOverride.scss [hunk 6]
```diff
@@ -0,0 +1,37 @@
+@use "@styles/styles";
```
Comment (lipezaballa) [inline src/styles/globalOverride.scss:1]: we'll have to fix it with my override methods

**Review by lipezaballa — APPROVED** [review]
